modules/virt-using-wasp-agent-to-configure-higher-vm-workload-density.adoc

```diff
@@ -0,0 +1,272 @@
+$ oc adm policy add-scc-to-user -n wasp privileged -z wasp
+----
+. Deploy `wasp-agent`:
+.. Create a `DaemonSet` object as follows:
++
+[source,terminal]
+----
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+name: wasp-agent
+namespace: wasp
+labels:
+app: wasp
+tier: node
+spec:
+selector:
+matchLabels:
+name: wasp
+template:
+metadata:
+annotations:
+description: >-
+Configures swap for workloads
+labels:
+name: wasp
+spec:
+serviceAccountName: wasp
+hostPID: true
+hostUsers: true
+terminationGracePeriodSeconds: 5
+containers:
+- name: wasp-agent
+image: >-
+quay.io/openshift-virtualization/wasp-agent:latest
+imagePullPolicy: Always
+env:
+- name: "FSROOT"
+value: "/host"
+resources:
+requests:
+cpu: 100m
+memory: 50M
+securityContext:
+privileged: true
+volumeMounts:
+- name: host
+mountPath: "/host"
+volumes:
+- name: host
+hostPath:
+path: "/"
+priorityClassName: system-node-critical
+updateStrategy:
+type: RollingUpdate
+rollingUpdate:
+maxUnavailable: 10%
+maxSurge: 0
+status: {}
+----
+.. Get the latest URL from the RH portal {_link?_} and enter in the created `DaemonSet` object.
+. Configure the `kubelet` service to permit swap:
+.. Create a `KubeletConfiguration` file as follows:
++
+[source,terminal]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: KubeletConfig
+metadata:
+name: custom-config
+spec:
+machineConfigPoolSelector:
+matchLabels:
+pools.operator.machineconfiguration.openshift.io/worker: ''  # MCP
+#machine.openshift.io/cluster-api-machine-role: worker # machine
+#node-role.kubernetes.io/worker: '' # node
+kubeletConfig:
+failSwapOn: false
+evictionSoft:
+memory.available: "1Gi"
+evictionSoftGracePeriod:
+memory.available: "1m30s"
+----
+.. Run the following command:
++
+[source,terminal]
+----
+$ oc wait mcp worker --for condition=Updated=True
+----
+. Create a `MachineConfig` object to provision swap as follows:
++
+[source,terminal]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+labels:
+machineconfiguration.openshift.io/role: worker
+name: 90-worker-swap
+spec:
+config:
+ignition:
+version: 3.4.0
+systemd:
+units:
+- contents: |
+[Unit]
+Description=Provision and enable swap
+ConditionFirstBoot=no
+
+          [Service]
+          Type=oneshot
+          Environment=SWAP_SIZE_MB=5000
+          ExecStart=/bin/sh -c "sudo dd if=/dev/zero of=/var/tmp/swapfile count=$SWAP_SIZE_MB bs=1MiB && sudo chmod 600 /var/tmp/swapfile && sudo mkswap /var/tmp/swapfile && sudo swapon /var/tmp/swapfile && free -h"
+
+          [Install]
+          RequiredBy=kubelet-dependencies.target
+        enabled: true
+        name: swap-provision.service
+----
++
+To have enough swap for the worst-case scenario, make sure to have at least as much swap space provisioned as overcommitted RAM. Calculate the amount of swap space to be provisioned on a node using the following formula:
++
+[source,terminal]
+----
+NODE_SWAP_SPACE = NODE_RAM * MEMORY_OVER_COMMIT_RATIO
```

Comment:
We will need to update the formula to: 
```
NODE_SWAP_SPACE = NODE_RAM * (MEMORY_OVER_COMMIT_PERCENT / 100% - 1)
```
As per the updated [doc](https://github.com/openshift-virtualization/wasp-agent/blob/main/docs/configuration.md).